specs/multicloud/Huawei_Cloud_File_Block.md

```diff
@@ -0,0 +1,931 @@
+Author(s): [Ashit Kumar](https://github.com/kumarashit), [Himanshu Varshney](https://github.com/himanshuvar)
+
+## Introduction
+SODA Foundation multi-cloud (Gelato) project provides policy based data mobility across multi-clouds. It supports public and Hybrid clouds. Gelato supports different types of storages like Object, File and Block. The supported cloud vendors are AWS, Azure, GCP, Huawei, IBM, Ceph, YIG-Ceph and Alibaba.
+
+Currently it provides solution for Object Migration, Policy Based Lifecycle Management of Objects in-cloud as well as Cross-cloud. It also provided File and Block Management operations.
+
+
+## Motivation and Background
+As of now, the supportability matrix for Vendors and Features stands as:
+
+![multicloud matrix](resources/multicloudMatrix.png)
+
+To create an unified mulitcloud management solution, there is need to support different cloud vendors for different services.
+Huawei Cloud being one of the major cloud providers for Hybrid Cloud Solution, support for the Fileshare and Disk service alongwith the currently supported object storage will add value.
+
+### Huawei Cloud Scalable File Service
+
+Scalable File Service (SFS) provides reliable, high-performance shared file storage hosted on HUAWEI CLOUD. With SFS, you can enjoy shared file access spanning multiple Elastic Cloud Servers (ECSs), Bare Metal Servers (BMSs), and containers created on Cloud Container Engine (CCE) and Cloud Container Instance (CCI).
+
+### Huawei Cloud Elastic Volume Service
+
+Elastic Volume Service (EVS) provides persistent block storage for services such as Elastic Cloud Server (ECS) and Bare Metal Server (BMS). With advanced data redundancy and cache acceleration capabilities, EVS offers high availability and durability.
+
+## GOALS
+- Extend support for Huawei Cloud
+- Add SFS to multi-cloud File service
+- Add EVS to multi-cloud Block service
+- Provide API interface
+- Provide any installation support for the services to be available
+
+## Non Goals
+- Only CRUD operations are scoped under this
+- Other operations like mount or any operations related to compute is not scoped as part of this
+- Only SFS is supported, no SFS-Turbo support
+
+## Requirement analysis
+### Input requirement
+- Provide support for Huawei Cloud SFS
+- Provide support for Huawei Cloud EVS
+
+### Feature Requirements
+- Add basic CRUD operations for Huawei Cloud SFS
+- Add basic CRUD Operations for Huawei Cloud EVS
+- These features form the building block for other features to be extended for Hybrid Cloud support
+
+## Architecture 
+### Here is the high level architecture for multicloud
+![multicloudArch](resources/MulticloudArcheditable.png)
+
+### Here is the high level design for File service
+![Module Arch FileShare](resources/MulticloudFileModuleArch.png)
+
+### Here is the high level design for Block service
+![Module Arch BlockShare](resources/MulticloudBlockService.png)
+
+## Interface Model
+### API
```

Comment:
for all the API, please add the return codes for non happy path cases...example - cannot create volume, volume to be deleted does not exist, in each API

Reply:
Yes, this we will put as part of Swagger file update